### DISCUSSION TITLE
[Reconciller] Why do I need to get the object in the cluster?

### DISCUSSION THREAD

**Discussion by rodrigovb96** [kubernetes-sigs/kubebuilder#3546] [Support] [answered]
Hi, I'm having some trouble to understand the role/logic a Reconciler.
My question is why the Reconcile only receives the name & namespace of the resource instead of the object.

Is it to prevent race conditions? or to follow the idea that the Reconciler should always look at the current state of the cluster when receiving a reconcile request?

In the same context how can access the underlying informer within the Reconciler context?

**Comment by camilamacedo86** [comment, marked as answer]
The doc added in the PR: https://github.com/kubernetes-sigs/kubebuilder/pull/3596
should answer your question. Please, check it out.

If you still qith questions feel free to open other issues.
